elastic/beats
# Handle agent configs

## What does this PR do?

Adds support for Fleet/Standalone configs from Elastic agent. This assumes a single stream (per the synthetics package) and uses that for the config, preferring `input.id` for the ID, since this is a solid UUID from Kibana tied to a consistent saved object.

Example `elastic-agent.yml`:

```yaml
outputs:
  default:
    type: elasticsearch
    hosts: [127.0.0.1:9200]
    username: elastic
    password: changeme

inputs:
  - id: cd8256b0-e86a-11ea-9199-e12923f35450
    name: synthetics-1
    type: synthetics/http
    use_output: default
    meta:
      package:
        name: synthetics
        version: 0.1.3
    data_stream:
      namespace: default
    streams:
      - name: fwae
        type: http
        data_stream:
          dataset: synthetics.monitor
          type: logs
        urls: feaw
        schedule: '@every 10s'
```

## Why is it important?

Requirement for https://github.com/elastic/uptime/issues/235

## Checklist


## Author's Checklist

- [ ] Check that it works with sample config and Fleet package

## How to test this PR locally

Use the example config at top. Check that the following dev console command returns new docs:

```
GET .ds-logs-elastic.agent.heartbeat-default-000001/_search
{
  "sort": {"@timestamp": "desc"}
}
```

## Related issues

Main Uptime Fleet issue: https://github.com/elastic/uptime/issues/235
